### PR TITLE
SALTO-5888 Jira DC: Fail on API-config with JSM on

### PR DESCRIPTION
**Pull request by kami-delaroz**
SALTO-5888 Jira DC: Fail on API-config with JSM on

---

_Additional context for reviewer_
None

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

**Review by shayc331 — COMMENTED** [review]
I am not sure if this is the change we need to solve it. IIUC The user experience does not change much after this pr, did you check how it looks from the UI side?
IMO the change should be in the SAAS, we need to remove the jsm checkbox when connecting new jira DC service. lets talk f2f

File: packages/jira-adapter/src/adapter_creator.ts [hunk 1]
```diff
@@ -39,6 +39,11 @@ function validateConfig(config: Values): asserts config is JiraConfig {
   Object.values(getApiDefinitions(apiDefinitions)).forEach(swaggerDef => {
     validateSwaggerApiDefinitionConfig('apiDefinitions', swaggerDef)
   })
+  if(isDataCenter) {
+    if(fetch.enableJSM || fetch.enableJSMPremium || fetch.enableJsmExperimental) {
```
Comment (shir-reifenberg) [inline packages/jira-adapter/src/adapter_creator.ts:43]: nit: extract to a function in the spirit of `isDCWithJsmEnabled()`

File: packages/jira-adapter/src/adapter_creator.ts [hunk 2]
```diff
@@ -39,6 +39,11 @@ function validateConfig(config: Values): asserts config is JiraConfig {
...
+      throw new Error('JSM is not supported for Jira DC')
```
Comment (shir-reifenberg) [inline packages/jira-adapter/src/adapter_creator.ts:44]: 1. Can we please add a log?
2. This is a user visible error. I think the error should be a little more informative and guide the user how to resolve it

File: packages/jira-adapter/src/adapter_creator.ts [hunk 3]
```diff
@@ -53,13 +58,14 @@ function validateConfig(config: Values): asserts config is JiraConfig {
...
   defaultConfig: JiraConfig,
+  isDataCenter: boolean,
 ): JiraConfig => {
   const configWithoutFetch = mergeWithDefaultConfig(
     _.omit(defaultConfig, 'fetch'),
     _.omit(config?.value ?? {}, 'fetch'),
   )
   const fetch = _.defaults({}, config?.value.fetch, defaultConfig.fetch)
-  const fullConfig = { ...configWithoutFetch, fetch }
+  const fullConfig = { ...configWithoutFetch, fetch, isDataCenter }
```
Comment (shir-reifenberg) [inline packages/jira-adapter/src/adapter_creator.ts:68]: why are we assigning this to the config ? I find it a little strange to assign it to the config when there is no `isDataCenter` field in it
I think we should just add `isDataCenter` as an additional param

**Review by shir-reifenberg — COMMENTED** [review]
Added some comments 🙃

**Comment by shir-reifenberg** [comment]
> I am not sure if this is the change we need to solve it. IIUC The user experience does not change much after this pr, did you check how it looks from the UI side? IMO the change should be in the SAAS, we need to remove the jsm checkbox when connecting new jira DC service. lets talk f2f

@shayc331 IIRC the error is visible (though not in an ideal way)

File: packages/jira-adapter/src/adapter_creator.ts (hunk 1 above)
Comment (kami-delaroz) [reply packages/jira-adapter/src/adapter_creator.ts:43]: these JSM flags are redundant and I'm planning to get rid of soon

**Comment by kami-delaroz** [comment]
@shayc331 @shir-reifenberg indeed the error is visible

File: packages/jira-adapter/src/adapter_creator.ts [hunk 4]
```diff
@@ -39,6 +39,14 @@ function validateConfig(config: Values): asserts config is JiraConfig {
   Object.values(getApiDefinitions(apiDefinitions)).forEach(swaggerDef => {
     validateSwaggerApiDefinitionConfig('apiDefinitions', swaggerDef)
   })
+  if (isDataCenter) {
+    if (fetch.enableJSM || fetch.enableJSMPremium || fetch.enableJsmExperimental) {
+      log.error('JSM is not supported in Jira DC config')
+      throw new Error(
+        'Failed to load Jira config. JSM is not supported for Jira DC, please remove enableJSM flag from the config file and try again',
```
Comment (shir-reifenberg) [inline packages/jira-adapter/src/adapter_creator.ts:46]: I think the part of `Failed to load Jira config` is not really user friendly and can probably be removed
Also we should probably point to https://help.salto.io/en/articles/7439324-salto-adapter-configuration-file

**Review by shir-reifenberg — APPROVED** [review]
LGTM! 🔥 
Can you please add a test covering this ?

File: packages/jira-adapter/src/adapter_creator.ts (hunk 4 above)
Comment (kami-delaroz) [reply packages/jira-adapter/src/adapter_creator.ts:46]: I copied it from NS validation error :)
but the config doc is a good idea

**Pull request merged**